dotnet/src/Extensions/Planning.StepwisePlanner/StepwisePlanner.cs

```diff
@@ -98,92 +127,109 @@ public Plan CreatePlan(string goal)
     public async Task<SKContext> ExecutePlanAsync(
         [Description("The question to answer")]
         string question,
-        [Description("List of tool descriptions")]
-        string functionDescriptions,
         SKContext context)
     {
+        if (string.IsNullOrEmpty(question))
+        {
+            context.Variables.Update("Question not found.");
+            return context;
+        }
+
         var stepsTaken = new List<SystemStep>();
-        if (!string.IsNullOrEmpty(question))
+
+        var promptRendered = new PromptTemplateEngine();
+        var chatHistory = new ChatHistory();
+
+        // TODO why doesn't context from method call work here
+        // TODO better error message if they don't include time.
+        var systemMessage = await promptRendered.RenderAsync(this._systemPrompt, this._context).ConfigureAwait(false);
+
+        chatHistory.AddSystemMessage(systemMessage);
+        if (this._examples != null)
         {
-            for (int i = 0; i < this.Config.MaxIterations; i++)
+            foreach (FewShotExample example in this._examples)
             {
-                var scratchPad = this.CreateScratchPad(question, stepsTaken);
+                var assistantMessage = await promptRendered.RenderAsync(example.Content, this._context).ConfigureAwait(false);
+                var authorRole = example.Role == Role.User ? AuthorRole.User : AuthorRole.Assistant;
 
-                context.Variables.Set("agentScratchPad", scratchPad);
+                chatHistory.AddMessage(authorRole, assistantMessage);
+            }
+        }
 
-                var llmResponse = (await this._systemStepFunction.InvokeAsync(context).ConfigureAwait(false));
+        chatHistory.AddUserMessage(question);
 
-                if (llmResponse.ErrorOccurred)
-                {
-                    throw new PlanningException(PlanningException.ErrorCodes.UnknownError, $"Error occurred while executing stepwise plan: {llmResponse.LastException?.Message}", llmResponse.LastException);
-                }
+        for (int i = 0; i < this.Config.MaxIterations; i++)
+        {
+            // TODO request settings and cancellation token
+            IReadOnlyList<IChatResult> llmResponse = await this._chatCompletion.GetChatCompletionsAsync(chatHistory).ConfigureAwait(false);
+            IChatResult llmResult = llmResponse.Single(); // TODO, is there a case where we get multiple messages?
+            ChatMessageBase responseMessage = await llmResult.GetChatMessageAsync().ConfigureAwait(false);
 
-                string actionText = llmResponse.Result.Trim();
-                this._logger?.LogTrace("Response: {ActionText}", actionText);
+            string actionText = responseMessage.Content.Trim();
```

Comment:
Another note for comparison -- I found it beneficial to not give the raw response as the message, but instead the parsed version (if valid) to help define the expectations better.